### DISCUSSION TITLE
Is there any performance advantage to minimizing custom macro use?

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [KaTeX/KaTeX#3642] [Q&A]
Title says it all: I'm in the habit of defining a lot of custom macros to facilitate styling (e.g. `\blue{Foo}` instead of `\htmlStyle{color:blue, font-weight:bold}{foo}`; however, I now have an option of using JS to generate the latter, as an alternative to such custom macros, so I am wondering if there is any performance-related benefit to such an approach?

**Comment by JacobBeaudry** [comment]
@edemaine if you have a sec to share your thoughts on this, I’d appreciate your insight! Is there any benefit to minimizing the use of custom macros that call other macros, in favour of JS functions that render the equivalent code, directly, at compile time?

**Reply by edemaine** [reply]
I think you'd need to benchmark to see if the difference is significant. JavaScript macros are probably slightly more efficient than LaTeX macros because they don't have to invoke the tokenizer. I doubt it'd be a big difference, though, compared e.g. to the browser rendering time.
